packages/bart/src/index.ts

```diff
@@ -0,0 +1,626 @@
+      <p>${trial_text.take_break_message}</p>
+      </div>`;
+    },
+   choices: [trial_text.continue_button],
+   button_html: (choice: string) => `<button class="jspsych-btn continue-button">${choice}</button>`,
+   on_load: function() {
+     if (enable_tts) {
+       const instructionsContent = document.querySelector('.instructions-container');
+       if (instructionsContent) {
+         const pageText = extractTextFromHtml(instructionsContent.innerHTML);
+         if (pageText.trim()) {
+           speakText(pageText, ttsOptions);
+         }
+       }
+     }
+   },
+   on_finish: function(data: any) {
+     stopAllSpeech();
+     data.task = 'bart';
+     data.phase = 'block-break';
+   }
+ }
+ return instructions;
+  }
+
+      //generate end results
+   function showEndResults(jsPsych: JsPsych, enable_tts = false, ttsOptions = {}) {
+
+    let USDollar = new Intl.NumberFormat('en-US', {
+          style: 'currency',
+          currency: 'USD'
+      });
+    
+   // console.log("showEndResults called");
+       const data = jsPsych.data.get().filter({ task: 'bart' });
+       const totalPoints = data.filter({ exploded: false, cashed_out: true }).select('pump_count').sum();
+
+     const stimulus = 
+     `<div class="instructions-container">
+      <p>${trial_text.final_earnings_message} <strong>${USDollar.format(totalPoints * 0.01)}</strong>!</p>
+      <p>${trial_text.thanks_message}</p>
+      </div>`;
+
+  const instructions = {
+    type: HtmlButtonResponsePlugin,
+    
+     stimulus: () => {
+       return stimulus;
+     },
+    choices: [trial_text.finish_button],
+    button_html: (choice: string) => `<button class="jspsych-btn continue-button">${choice}</button>`,
+    on_load: function() {
+      if (enable_tts) {
+        const instructionsContent = document.querySelector('.instructions-container');
+        if (instructionsContent) {
+          const pageText = extractTextFromHtml(instructionsContent.innerHTML);
+          if (pageText.trim()) {
+            speakText(pageText, ttsOptions);
+          }
+        }
+      }
+    },
+    on_finish: function(data: any) {
+      stopAllSpeech();
+      data.task = 'bart';
+      data.phase = 'end-results';
+    }
+  }
+  return instructions;
+   }
+
+
+
+function createTrialTimeline(jsPsych: JsPsych, max_pumps: number, min_pumps: number, currency_unit_per_pump: number, trial_timeout?: number, enable_timeout?: boolean) {
+    
+// const explosion_range = max_pumps - min_pumps;
+//       //const explosion_point = Math.floor(Math.random() * explosion_range) + MIN_PUMPS;
+//     const explosion_point = 10; // fixed explosion point for testing
+//      let pump_count = 0;
+//      let balloon_popped = false;
+//      let cashed_out = false;
+
+
+    let pump_count: number;
+    let balloon_popped: boolean;
+    let cashed_out: boolean;
+    let timed_out: boolean;
+    let explosion_point: number;
+
+     let USDollar = new Intl.NumberFormat('en-US', {
```

Comment:
seems like duplicate code? needed here? maybe a way to move this somewhere so it can be shared?